pkg/serverless/aws/arn.go

```diff
@@ -55,17 +81,42 @@ func SetRequestID(reqID string) {
 	defer currentReqID.Unlock()
 
 	currentReqID.value = reqID
+	cacheRequestID(reqID)
 }
 
-// FunctionNameFromARN returns the function name from the currently set ARN.
-// Thread-safe.
-func FunctionNameFromARN() string {
-	currentARN.Lock()
-	defer currentARN.Unlock()
-	if currentARN.value == "" {
+// cacheRequestID writes the Request ID to a file in the /tmp directory so we can restore it if the extension process is restarted
+func cacheRequestID(reqID string) {
```

Comment:
Is it a lot of work to save requestID and ARN in a single file? It seems like this approach might not scale if we want to save more metadata.

Reply:
I wouldn't say it's a lot more work, but it didn't seem necessary. It would just add complexity. Reading and writing a single value to/from a single file is dirt simple, if we have two values in a file then we have to think about parsing it, carefully editing the file as opposed to just nuking it when we get a new value, etc. If we end up wanting to save more metadata down the road, I figure we could revisit this then, but for now trying to get these into one file instead of two seemed premature.

Reply:
We could indeed go with only one file containing a JSON for instance (if this JSON struct could have a `version` field that would be awesome, just-in-case).